packages/generator/src/global-name-formatter.ts

```diff
@@ -41,25 +44,17 @@ export class GlobalNameFormatter {
   }
 
   private transformAndCacheDirectoryName(directoryName: string): string {
-    return this.transformAndCache(directoryName, this.directoryNamesCache);
+    const newName = this.directoryNameFinder.findUniqueName(directoryName);
+    this.directoryNameFinder.addToAlreadyUsedNames(newName);
```

Comment:
The beauty of the old solution was, that getting a new name was atomic, so you could not forget to cache the name. The uglyness was that the function had a side effect. With the new solution it is the other way around. I guess you considered that and in the end I believe we have to die one death ;)
I don't have a very strong opinion here, but this seems a little more dangerous now.

Reply:
I think it is fair that the finder does not cache. The name formatter does then automatially.

Reply:
I agree 😊